pkg/controllers/metrics/pod/controller.go

```diff
@@ -261,8 +261,11 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 
 func (c *Controller) recordPodSchedulingUndecidedMetric(pod *corev1.Pod) {
 	nn := client.ObjectKeyFromObject(pod)
-	// If we've made a decision on this pod, delete the metric idempotently and return
-	if decisionTime := c.cluster.PodSchedulingDecisionTime(nn); !decisionTime.IsZero() {
+	cond, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
+		return c.Type == corev1.PodScheduled
```

Comment:
nit: If we check that the status of PodScheduled is True here then the `ok` can be `scheduled` and we can use `|| scheduld` on line 268

Reply:
+1, I'd just short-circuit here and delete the metric up top just to reduce the number of conditionals

Reply:
Done